context_test.go

```diff
@@ -577,6 +578,128 @@ func TestOrigin(t *testing.T) {
 	}
 }
 
+// TestServerContextClose tests closing server using cancellable context, ensure that clients are disconnected
+//
+// Test:
+//
+//   - Start two blip contexts: an echo server and an echo client
+//   - The echo server is configured to respond to incoming echo requests and return responses
+//   - The echo client tries to read the response after sending the request
+//   - Expected: the echo client should receive some sort of error when trying to read the response, since the server abruptly terminated the connection
+//   - Actual: the echo client blocks indefinitely trying to read the response
+func TestServerContextClose(t *testing.T) {
+
+	blipContextEchoServer, err := NewContext(defaultContextOptions)
```

Comment:
As an API design, I think `NewContext` should look like `func NewContext(ctx context.Context, opts ContextOptions) (*Context, error)`

which will set the `cancelCtx` on `blip.Context` and force the callers to specify it, even if they decide to do `context.TODO()` or `context.Background()`.